{"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"NIOTLSUnwrappingError"}],"languages":["swift"],"platforms":["macOS"]}]}],"schemaVersion":{"major":0,"minor":3,"patch":0},"sections":[],"variants":[{"paths":["\/documentation\/niossl\/niotlsunwrappingerror"],"traits":[{"interfaceLanguage":"swift"}]}],"relationshipsSections":[{"identifiers":["doc:\/\/org.swift.docc\/SQ","doc:\/\/org.swift.docc\/s5ErrorP","doc:\/\/org.swift.docc\/SH","doc:\/\/org.swift.docc\/s8SendableP"],"kind":"relationships","title":"Conforms To","type":"conformsTo"}],"identifier":{"url":"doc:\/\/org.swift.docc\/documentation\/NIOSSL\/NIOTLSUnwrappingError","interfaceLanguage":"swift"},"abstract":[{"type":"text","text":"Represents errors that may occur while attempting to unwrap TLS from a connection."}],"kind":"symbol","metadata":{"fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"NIOTLSUnwrappingError"}],"title":"NIOTLSUnwrappingError","roleHeading":"Enumeration","role":"symbol","symbolKind":"enum","externalID":"s:6NIOSSL21NIOTLSUnwrappingErrorO","modules":[{"name":"NIOSSL"}],"navigatorTitle":[{"kind":"identifier","text":"NIOTLSUnwrappingError"}]},"hierarchy":{"paths":[["doc:\/\/org.swift.docc\/documentation\/NIOSSL"]]},"topicSections":[{"title":"Enumeration Cases","identifiers":["doc:\/\/org.swift.docc\/documentation\/NIOSSL\/NIOTLSUnwrappingError\/alreadyClosed","doc:\/\/org.swift.docc\/documentation\/NIOSSL\/NIOTLSUnwrappingError\/closeRequestedDuringUnwrap","doc:\/\/org.swift.docc\/documentation\/NIOSSL\/NIOTLSUnwrappingError\/invalidInternalState","doc:\/\/org.swift.docc\/documentation\/NIOSSL\/NIOTLSUnwrappingError\/unflushedWriteOnUnwrap"]},{"title":"Default Implementations","identifiers":["doc:\/\/org.swift.docc\/documentation\/NIOSSL\/NIOTLSUnwrappingError\/Equatable-Implementations","doc:\/\/org.swift.docc\/documentation\/NIOSSL\/NIOTLSUnwrappingError\/Error-Implementations"],"generated":true}],"references":{"doc://org.swift.docc/documentation/NIOSSL/NIOTLSUnwrappingError/alreadyClosed":{"role":"symbol","title":"NIOTLSUnwrappingError.alreadyClosed","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"alreadyClosed"}],"abstract":[{"type":"text","text":"The TLS channel has already been closed, so it is not possible to unwrap it."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIOSSL\/NIOTLSUnwrappingError\/alreadyClosed","kind":"symbol","type":"topic","url":"\/documentation\/niossl\/niotlsunwrappingerror\/alreadyclosed"},"doc://org.swift.docc/SH":{"type":"unresolvable","title":"Swift.Hashable","identifier":"doc:\/\/org.swift.docc\/SH"},"doc://org.swift.docc/documentation/NIOSSL/NIOTLSUnwrappingError/Equatable-Implementations":{"role":"collectionGroup","title":"Equatable Implementations","abstract":[],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIOSSL\/NIOTLSUnwrappingError\/Equatable-Implementations","kind":"article","type":"topic","url":"\/documentation\/niossl\/niotlsunwrappingerror\/equatable-implementations"},"doc://org.swift.docc/documentation/NIOSSL/NIOTLSUnwrappingError/Error-Implementations":{"role":"collectionGroup","title":"Error Implementations","abstract":[],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIOSSL\/NIOTLSUnwrappingError\/Error-Implementations","kind":"article","type":"topic","url":"\/documentation\/niossl\/niotlsunwrappingerror\/error-implementations"},"doc://org.swift.docc/documentation/NIOSSL/NIOTLSUnwrappingError/invalidInternalState":{"role":"symbol","title":"NIOTLSUnwrappingError.invalidInternalState","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"invalidInternalState"}],"abstract":[{"type":"text","text":"The internal state of the handler is not able to process the unwrapping request."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIOSSL\/NIOTLSUnwrappingError\/invalidInternalState","kind":"symbol","type":"topic","url":"\/documentation\/niossl\/niotlsunwrappingerror\/invalidinternalstate"},"doc://org.swift.docc/documentation/NIOSSL":{"role":"collection","title":"NIOSSL","abstract":[],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIOSSL","kind":"symbol","type":"topic","url":"\/documentation\/niossl"},"doc://org.swift.docc/SQ":{"type":"unresolvable","title":"Swift.Equatable","identifier":"doc:\/\/org.swift.docc\/SQ"},"doc://org.swift.docc/s5ErrorP":{"type":"unresolvable","title":"Swift.Error","identifier":"doc:\/\/org.swift.docc\/s5ErrorP"},"doc://org.swift.docc/documentation/NIOSSL/NIOTLSUnwrappingError":{"role":"symbol","title":"NIOTLSUnwrappingError","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"NIOTLSUnwrappingError"}],"abstract":[{"type":"text","text":"Represents errors that may occur while attempting to unwrap TLS from a connection."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIOSSL\/NIOTLSUnwrappingError","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"NIOTLSUnwrappingError"}],"url":"\/documentation\/niossl\/niotlsunwrappingerror"},"doc://org.swift.docc/documentation/NIOSSL/NIOTLSUnwrappingError/closeRequestedDuringUnwrap":{"role":"symbol","title":"NIOTLSUnwrappingError.closeRequestedDuringUnwrap","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"closeRequestedDuringUnwrap"}],"abstract":[{"type":"text","text":"We were unwrapping the connection, but during the unwrap process a close call"},{"type":"text","text":" "},{"type":"text","text":"was made. This means the connection is now closed, not unwrapped."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIOSSL\/NIOTLSUnwrappingError\/closeRequestedDuringUnwrap","kind":"symbol","type":"topic","url":"\/documentation\/niossl\/niotlsunwrappingerror\/closerequestedduringunwrap"},"doc://org.swift.docc/s8SendableP":{"type":"unresolvable","title":"Swift.Sendable","identifier":"doc:\/\/org.swift.docc\/s8SendableP"},"doc://org.swift.docc/documentation/NIOSSL/NIOTLSUnwrappingError/unflushedWriteOnUnwrap":{"role":"symbol","title":"NIOTLSUnwrappingError.unflushedWriteOnUnwrap","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"unflushedWriteOnUnwrap"}],"abstract":[{"type":"text","text":"This write was failed because the channel was unwrapped before it was flushed."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIOSSL\/NIOTLSUnwrappingError\/unflushedWriteOnUnwrap","kind":"symbol","type":"topic","url":"\/documentation\/niossl\/niotlsunwrappingerror\/unflushedwriteonunwrap"}}}